app/src/main/java/org/simple/clinic/home/HomeScreenController.kt

```diff
@@ -39,7 +83,8 @@ class HomeScreenController @Inject constructor(
                 facilityRepository.currentFacility(it)
               }
         }
-        .map { { ui: Ui -> ui.setFacility(it.name) } }
+        .replay()
+        .refCount()
```

Comment:
The sharing of this stream should not be done here, but only in the call site where it is needed.

Reply:
Done